model/src/main/proto/survey.proto

```diff
@@ -0,0 +1,182 @@
+message SurveyUserAnswer{
+  //  The ID of the survey to which this answer is tied.
+  string  survey_id = 1;
+
+  // Semantic name of the question the response is tied to.
+  QuestionName question_name = 2;
+
+  // The value of the answer the user gave.
+  string user_answer = 3;
+}
+
+// Represents the different types of users who could be taking the survey.
+// Consists of answer options for the user type question.
+enum  UserTypeAnswer{
+  // User type unknown.
+  USER_TYPE_UNSPECIFIED =0;
+  // Corresponds to an unspecified or unknown type of user.
+  OTHER = 1;
+
+  // Corresponds to a user who is a teacher.
+  TEACHER = 2;
+
+  // Corresponds to a user who is a parent.
+  PARENT = 3;
+
+  // Corresponds to a user who is a learner.
+  LEARNER = 4;
+}
+
+// Represents general opinion of the user about the Oppia android app.
+// Consists of answer options for the market fit question.
+enum  MarketFitAnswer{
+  // Answer unknown.
+  MARKET_FIT_ANSWER_UNSPECIFIED =0;
+
+  // Corresponds to the answer for Very Disappointed.
+  VERY_DISAPPOINTED = 1;
+
+  // Corresponds to the answer for Disappointed.
+  DISAPPOINTED = 2;
+
+  // Corresponds to the answer for  Not Disappointed.
+  NOT_DISAPPONTED = 3;
+
+  // Corresponds to the answer for  N/A - Don't use Oppia Anymore.
+  NA_DONT_USE_OPPIA_ANYMORE = 4;
+}
+
+// Represents the status of the survey at the time it is exited.
+enum SurveyCompletionStatus{
+  // The completion status is unknown.
+  SURVEY_COMPLETION_STATUS_UNSPECIFIED = 0;
+
+  // Corresponds to the survey being declined, not started.
+  SURVEY_DECLINED = 1;
+
+  // Corresponds to the survey being exited before the mandatory questions are completed.
+  SURVEY_STARTED_NOT_COMPLETED = 2;
+
+  // Corresponds to the mandatory part of the survey being completed.
+  SURVEY_MANDATORY_PART_COMPLETED = 3;
+
+  // Corresponds to the optional part of the survey being completed.
+  SURVEY_COMPLETED = 4;
+}
+
+// Corresponds to a survey question that exists ephemerally in the UI and will disappear
+// once the user finishes or exits the survey session.
+message EphemeralSurveyQuestion{
+  // The actual question the user is currently viewing.
+  SurveyQuestion  question = 1;
+
+  //Indicates whether this question has a previous question that has been answered,
+  // or if this is the initial question.
+  bool has_previous_question = 2;
+
+  // Indicates whether this question has an upcoming question that can be navigated to.
+  bool has_next_question = 3;
+
+  //  Indicates possible states a question can take depending on whether a user needs to provide an
+  //  answer or has navigated to a previous question, or has reached the last question.
+  oneof question_type{
+    // A pending question that requires an answer to continue.
+    bool pending_question = 4;
+
+    // A previous question completed by the user.
+    CompletedQuestion completed_question = 5;
+
+    // True if this is the last question in the survey.
```

Comment:
Probably more like: "Present if this is the last question in the survey (the actual value of the boolean does not matter)."

We may set it to true, but just its presence as being set is the key indicator (per how we handle this with other bool oneof cases).

Reply:
Refactored the comment.